### PR TITLE
Password Reset and Email Validation

### PR DESCRIPTION
**Pull request by DnOberon**
## Description
Added email validation. User creation (standard not super user) will now send an email if validation is enforced. That email will contain a URL with a token attached. That token is then used in a GET request against the server to validate that email. 

Added password reset capability. Users can now request a 4 hour window in which to reset their password. An email will be sent to the user's email with a token and a time the token was issued. The token, new password, and email are then POSTed to the server.

## Motivation and Context
Both email validation and password reset are common user features needed to present a full-fledged product to the end user.

## How Has This Been Tested?
Unit tests written where appropriate. Testing via the UI and raw calls to the HTTP endpoints covered the rest.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: api_documentation/Core.postman_collection.json [hunk 1]
```diff
@@ -1795,46 +1795,6 @@
 							},
 							"response": []
 						},
-						{
-							"name": "Update Type Relationship Pair",
```
Comment (jerenmb) [inline api_documentation/Core.postman_collection.json:1799]: This deletion seems unrelated, did you mean to remove this API from the docs?

Comment (DnOberon) [reply api_documentation/Core.postman_collection.json:1799]: I believe this was a duplicate. I still have one for the route showing up.

**Review by jerenmb — APPROVED** [review]
